tests/test_generators.py

```diff
@@ -1822,6 +1822,63 @@ class Singular(Base):
             """,
         )
 
+    def use_inflect_pluralization(
+        self, generator: CodeGenerator, singular: str, plural: str
+    ) -> None:
+        Table(
+            "simple_items",
+            generator.metadata,
+            Column("id", INTEGER, primary_key=True),
+            Column(f"{singular}_id", INTEGER),
+            ForeignKeyConstraint([f"{singular}_id"], [f"{plural}.id"]),
+            UniqueConstraint(f"{singular}_id"),
+        )
+        Table(plural, generator.metadata, Column("id", INTEGER, primary_key=True))
+
+        validate_code(
+            generator.generate(),
+            f"""\
+from sqlalchemy import Column, ForeignKey, Integer
+from sqlalchemy.orm import declarative_base, relationship
+
+Base = declarative_base()
+
+
+class {singular.capitalize()}(Base):
+    __tablename__ = '{plural}'
+
+    id = Column(Integer, primary_key=True)
+
+    simple_item = relationship('SimpleItem', uselist=False, \
+back_populates='{singular}')
+
+
+class SimpleItem(Base):
+    __tablename__ = 'simple_items'
+
+    id = Column(Integer, primary_key=True)
+    {singular}_id = Column(ForeignKey('{plural}.id'), unique=True)
+
+    {singular} = relationship('{singular.capitalize()}', back_populates='simple_item')
+            """,
+        )
+
+    @pytest.mark.parametrize("generator", [["use_inflect"]], indirect=True)
+    def test_use_inflect_plural_s(self, generator: CodeGenerator) -> None:
+        self.use_inflect_pluralization(generator, "manufacturer", "manufacturers")
+
+    @pytest.mark.parametrize("generator", [["use_inflect"]], indirect=True)
+    def test_use_inflect_plural_es(self, generator: CodeGenerator) -> None:
+        self.use_inflect_pluralization(generator, "status", "statuses")
+
+    @pytest.mark.parametrize("generator", [["use_inflect"]], indirect=True)
+    def test_use_inflect_plural_ies(self, generator: CodeGenerator) -> None:
+        self.use_inflect_pluralization(generator, "study", "studies")
+
+    @pytest.mark.parametrize("generator", [["use_inflect"]], indirect=True)
+    def test_use_inflect_singular(self, generator: CodeGenerator) -> None:
+        self.use_inflect_pluralization(generator, "moose", "moose")
```

Comment:
This is not what I meant. I meant for you to use `@pytest.mark.parametrize` to supply those parameters.

Reply:
Implemented parameterized tests